### ISSUE TITLE
Blank Screen on iOS

### ISSUE THREAD

**Issue by oliverfogden** [jeduan/cordova-plugin-facebook4#267]
When trying to use this plugin on iOS and calling:

```
facebookConnectPlugin.login(['public_profile','email'], function(data){
          console.log("FB",data)
        }, function(error){
          console.log("ERROR",error)
} )
```

The web view appears to have a blank screen slide up in front of it from the bottom edge of the screen. This screen is never filled and stays empty effectively bricking the app.

I was able to replicate this on the sample cordova app by adding the above code snippet into the receivedEvent function.

The only log messages shown are:
2016-07-18 16:18:44.232 HelloCordova[9594:5572008] Starting login
2016-07-18 16:18:44.297 HelloCordova[9594:5572008] THREAD WARNING: ['FacebookConnectPlugin'] took '65.583984' ms. Plugin should use a background thread.

On android the same code works as expected, bringing up the Facebook login.

Has anyone else experienced this before?

Installed platforms: android 5.1.1, iOS 4.1.0
cordova 6.1.0

**Comment by oliverfogden** [comment]
I just thought I would note, the above is testing on a real device (iPhone 6S - iOS 9.3.2). When testing on the the simulator, the plugin seems to be working and it pulls up a Facebook login through the inappbrowser

**Comment by Vantablack** [comment]
@oliverfogden  Not sure if you're using Ionic 2 but here's how I solved this issue for me.

Try importing `Facebook` from `ionic-native`. Source: [Ionic Native Doc](http://ionicframework.com/docs/v2/native/facebook/)

```typescript
import {Facebook} from 'ionic-native';
```

Then you can simply use `Facebook.login(...)`.
Note that `login()` is a **static** function.
```typescript
Facebook.login(["public_profile", "email", "user_friends"])
  .then(function(success) {
    // { id: "634565435",
    //   lastName: "bob"
    //   ...
    // }
  }, function (error) {
    // error
});
```

You might also want to take a look at the [source code](https://github.com/driftyco/ionic-native/blob/master/src/plugins/facebook.ts#L119) for `facebook.ts`.


```bash
$ ionic info

Your system information:

Cordova CLI: 6.2.0
Gulp version:  CLI version 1.2.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.10
Ionic CLI Version: 2.0.0-beta.32
Ionic App Lib Version: 2.0.0-beta.18
ios-deploy version: 1.8.6
ios-sim version: 5.0.8
OS: Mac OS X El Capitan
Node Version: v5.10.0
Xcode version: Xcode 7.3 Build version 7D175
```

iOS Version: 9.0.2

**Comment by oliverfogden** [comment]
Hello!

Thanks for your message @Vantablack, but I'm not using ionic. I do normally get the thread warning though!

For some obscure reason, I ended up seeing the same initial issue (blank page sliding up) happening on another app (Tinder!). I ended up downloading the example Facebook login app (native) and installing it on my iPhone, seemingly from then on the problem seems to have disappeared.

Fingers crossed this was some really bizarre bug **_somewhere_** never to be seen again...
